spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowServerConfiguration.java

```diff
@@ -307,6 +302,18 @@ public TaskRepositoryInitializer taskRepositoryInitializerForDefaultDB(DataSourc
 			return taskRepositoryInitializer;
 		}
 
+		@Bean
+		public DefinitionRepositoryInitializer definitionRepositoryInitializerForDefaultDB(DataSource dataSource, Server server) {
```

Comment:
Good question.  In short when using the embedded h2 server (default) we have to wait for the H2 server to be available prior to trying to connect to it.  When not using the embedded server we use the other definitionRepositoryInitializerForDefaultDB (No server param) to connect to an external datasource.